dockers/docker-fpm-frr/bgpcfgd

```diff
@@ -290,6 +292,34 @@ class BGPPeerMgr(Manager):
         vrf, nbr = key.split('|', 1)
         if key not in self.peers:
             cmd = None
+
+            if "local_addr" not in data:
+                syslog.syslog(syslog.LOG_WARNING, 'Peer {}. Error in missing required attribute "local_addr"'.format(key))
+            else:
+                # The route that belongs to a vnet cannot be advertised by the default BGP session.
+                # So we need to check whether this route belongs to a vnet.
+                local_addresses = self.directory.get_slot("local_addresses")
+                # Check if the information for the local address of this route has been set
+                # Which means we can find the interface for this route by its local address
+                # We can only process this route message when the local address was set
+                if data["local_addr"] not in local_addresses:
+                    return False
+                local_address = local_addresses[data["local_addr"]]
+                interfaces = self.directory.get_slot("interfaces")
+                # Check if the information for the interface of this local address has set
+                # Which means we can determine whether this route belongs to a vnet
+                # Because the vnet name will be set in the interface
+                # Ref:https://github.com/Azure/SONiC/blob/master/doc/vxlan/Vxlan_hld.md#212-vnetinterface-table
+                if local_address.has_key("interface") and local_address["interface"] in interfaces:
+                    interface = interfaces[local_address["interface"]]
+                    # Check if this route is belong to a vnet
+                    if interface.has_key("vnet_name") and interface["vnet_name"]:
+                        # Ignore the route that is in a vnet
+                        syslog.syslog(syslog.LOG_INFO, 'Peer {} in vnet {}'.format(key, interface["vnet_name"]))
+                        return True
```

Comment:
i finally understood what you are trying to do in this block (spend about 30 minutes)! it is so difficult to understand because people do not understand what local_addresses and interfaces mean. In fact, what you really should do here is to create two functions.
one function is 
find_local_interface(ip). The input is a ip address, and return value is interface name. return None if you cannot find a interface.

second function is
check_interface_type(interface_name). You can return vnet interface it is vnet interface, or default if the interface is in default vrf.

since you do not have such abstract, it is difficult for others to understand the code and maintain. 

Please put code readability into priority.